nmdc_runtime/api/core/metadata.py

```diff
@@ -37,11 +37,14 @@
     if "items" in db_prop_spec and "$ref" in db_prop_spec["items"]
 }
 
-collection_names_with_unique_id = {
-    coll_name
-    for coll_name, class_name in collection_name_to_class_name.items()
-    if "id" in get_nmdc_jsonschema_dict()["$defs"][class_name].get("required", [])
-}
+
+@lru_cache
```

Comment:
Do we need to worry about cache invalidation? Or would  that only happen if a new version is deployed.

Reply:
right, we explicitly update the `nmdc-schema` PyPI package dependency, and that necessitates a fresh deployment, so `get_nmdc_jsonschema_dict()` can be cached on init for a deployment.